### PR TITLE
Messaging session bug fix

### PR DESCRIPTION
**Pull request by manasisurve**
**Issue #:**
 https://github.com/aws/amazon-chime-sdk-js/issues/2372

**Description of changes:**
1. Fix to break infinite reconnect loop in messagingSession on timeout. Reported here https://github.com/aws/amazon-chime-sdk-js/issues/2372
2. Add backwards compatibility with aws-js-sdk 2.x for getMessagingSessionEndpoint call

**Testing:**
* Tested reconnect loop fix by expiring credentials to validate reconnect loop on messagingSession breaks after the timeout
* Tested compatibility with aws sdk v2 by changing the chime client instantiation in the messagingSession demo

* Set aws credentials with in correct aws session token
* Run the messagingSession demo here: https://github.com/aws/amazon-chime-sdk-js/tree/main/demos/browser. npm run start --app=messagingSession
* connect with an app instance user
* Reconnect loop is triggered but breaks after timeout

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
*To Test aws js sdk v2 compatibility run the messagingSession demo with below changes*
 Make the following changes in amazon-chime-sdk-js/demos/browser/app/messagingSession/messagingSession.ts
```
import { ChimeSDKMessagingClient } from '@aws-sdk/client-chime-sdk-messaging'
to
import * as Chime from 'aws-sdk/clients/chime';
```

and change 
```
 const chime = new ChimeSDKMessagingClient({ region: 'us-east-1', credentials: response });
to
const chime = new Chime({ region: 'us-east-1', credentials: response });
```

*To Test aws js sdk v3 with v2 style run the messagingSession demo with below changes*
 Make the following changes in amazon-chime-sdk-js/demos/browser/app/messagingSession/messagingSession.ts
```
import { ChimeSDKMessagingClient } from '@aws-sdk/client-chime-sdk-messaging'
to
import { ChimeSDKMessaging } from '@aws-sdk/client-chime-sdk-messaging';
```
and 
```
 const chime = new ChimeSDKMessagingClient({ region: 'us-east-1', credentials: response });
to
const chime = new ChimeSDKMessaging({ region: 'us-east-1', credentials: response });
```

and then run 
**Checklist:**

1. Have you successfully run `npm run build:release` locally?
yes

3. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
no

4. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
no
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
yes

### CODE REVIEW

**Review by dpwspoon — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- `MessagingSession` infinite reconnect loop to break on timeout.
+- `MessagingSession` `getMessagingSessionEndpoint` call is now backwards compatible with aws js sdk v2 and lower.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:21]: nit: aws js sdk v2 -> AWS JS SDK v2. 

Should we drop "and lower"? Not sure if we support it.

File: src/messagingsession/DefaultMessagingSession.ts [hunk 2]
```diff
@@ -108,12 +108,31 @@ export default class DefaultMessagingSession implements MessagingSession {
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
     let endpointUrl = this.configuration.endpointUrl;
+
+    if (!reconnecting) {
```
Comment (devalevenkatesh) [inline src/messagingsession/DefaultMessagingSession.ts:112]: So any reason for moving this up?

Comment (manasisurve) [reply src/messagingsession/DefaultMessagingSession.ts:114]: Updated comment to reflect the reason

File: CHANGELOG.md (hunk 1 above)
Comment (manasisurve) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- `MessagingSession` reconnect loop did not break on error past reconnect deadline. Infinite reconnect loop was caused due to firstConnectionAttemptTimestamp not being set as startedConnectionAttempt was not invoked. Check https://github.com/aws/amazon-chime-sdk-js/issues/2372 for details.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:23]: nit: Back tick `firstConnectionAttemptTimestamp` and `startedConnectionAttempt`.

**Review by devalevenkatesh — APPROVED** [review]

**Review by devalevenkatesh — COMMENTED** [review]
Is the `package-lock.json` changes necessary?

Does not look like from the fix that those are needed.

**Comment by devalevenkatesh** [comment]
Closing due to https://github.com/aws/amazon-chime-sdk-js/pull/2444

**Pull request closed without merging**